Adds distributed UNet example

dmsuehir opened the pull request:
This PR adds an examples directory, with the distributed UNet use case as our first example.  I have a README that explains the steps on running the example (which involves using KVC and MLT) as well as information on the steps that I took to go from an MLT template to running the UNet model.

## Reviewer Checklist

 - [x] Do you understand it?
 - [x] Is it correct?
 - [x] Is it safe?
 - [x] Is it legally compliant?
 - [x] Is it robust?
 - [x] Is it simple?
 - [x] Is it tested?
 - [x] Is it documented?
 - [x] Is the style consistent?

See [the reviewer guide](docs/reviews.md) for more detail on each check.

karthikvadla commented:
One more thing, did you add K8s directory . It complied about that while deploying . Otherthan that everything looks good .

karthikvadla commented:
Just saw you already added k8s directory . Looks good to me .

dmsuehir commented on examples/distributed_unet/README.md:
```diff
@@ -0,0 +1,308 @@
+ parameter server, job name, and task index).
+
+The updated app is located in the [mlt_app](mlt_app) directory.  Before
+running the app, you will need to update:
```
Yes, I added the mlt.json.  thanks

dmsuehir commented:
@ashahba Should I add license headers to the UNet code files and remove the `LICENSE` file?  I grabbed the UNet model from here: https://github.com/NervanaSystems/topologies/tree/master/distributed_unet
Where they don't have the headers, but they have a `LICENSE` file in that directory.

ashahba commented:
@dmsuehir let me review and I get back to you on this.

nqn commented on examples/distributed_unet/README.md:
```diff
@@ -0,0 +1,308 @@
+- [Kube-Volume-Controller](https://github.com/kubeflow/experimental-kvc/blob/master/docs/ops.md#installing-the-controller)
+- [TF-Operator](https://github.com/kubeflow/tf-operator)
+
+You will also need to register to obtain the BraTS dataset.  Information
```
Nit: maybe move "Information" to the next line

nqn commented on examples/distributed_unet/README.md:
````diff
@@ -0,0 +1,308 @@
+volumemanager "unet-data" created
+```
+
+Once the data transfer has completed, the `volumemanager` has
````
'has `message: success`'? What does that mean :) in the status?

nqn commented on examples/distributed_unet/README.md:
````diff
@@ -0,0 +1,308 @@
+```
+
+Once the data transfer has completed, the `volumemanager` has
+`message: success`.  We can then get the `nodeAffinity` information,
````
Nit: could consider to drop 'information' so it sounds like "get the node affinity"

nqn commented on examples/distributed_unet/mlt_app/k8s-templates/tfjob.yaml:
```diff
@@ -0,0 +1,92 @@
+                  - key: kubernetes.io/hostname
+                    operator: In
+                    values:
+                    - gke-dls-us-n1-standard-4-1ba3a893-516b
```
These are internal server names - can we replace them with placeholders?

nqn commented on examples/distributed_unet/mlt_app/k8s-templates/tfjob.yaml:
```diff
@@ -0,0 +1,92 @@
+                  - key: kubernetes.io/hostname
+                    operator: In
+                    values:
+                    - gke-dls-us-n1-highmem-8-skylake-82af83b4-7gc4
```
Same here ^^

nqn commented on examples/distributed_unet/mlt_app/model.py:
```diff
@@ -0,0 +1,175 @@
+import os.path
```
Do we need license headers here?

nqn commented on examples/distributed_unet/mlt_app/preprocess.py:
```diff
@@ -0,0 +1,399 @@
+""" To conver dicom images into images needed for keras"""
```
Same here - license headers?

ashahba commented:
@dmsuehir If we changes any of the files in:
https://github.com/NervanaSystems/topologies open to public?

Then for the files we changed, we need to have both license headers (in case the licenses are different). Ours on top followed by the original header.

And technically we don't need to keep that `LICENSE` since we already have a exact copy of it shipped with our code so end users already have one.

But based on what's under `Citations` for that repo, if we used `BraTS` data, we need to cite it.

Also as @nqn mentioned above we need to make server names anonymized like for example:
```gke-mycluster-mynode-82af83b4-7gc4```

dmsuehir commented:
Thanks @ashahba.  I'll add the license header and will remove the `LICENSE` file.  I have the BraTS Citation already -- at the bottom of the `README.md`.

dmsuehir commented:
@ashahba @nqn  I made the updates you requested.  Added license headers and modified the server names to be like `gke-cluster-n1-highmem-8-skylake-1`, `gke-cluster-n1-highmem-8-skylake-2`, etc., and made README file updates suggested by Niklas (including adding a link to the example from the main README).
